tests/rpmbuild.at

```diff
@@ -2272,3 +2272,27 @@ runroot rpmbuild \
 ],
 [ignore])
 AT_CLEANUP
+
+# ------------------------------
+# Check if dynamic spec generation works
+AT_SETUP([rpmbuild with dynamic spec generation])
+AT_KEYWORDS([build])
+RPMDB_INIT
+AT_CHECK([
+
+run rpmbuild --noclean\
+  -ba "${abs_srcdir}"/data/SPECS/dynamic.spec
```

Comment:
Use a simpler, noarch package that you can run under runroot instead, what's being tested here doesn't require a compiler and make to be present at all. It makes things simpler when you're always operating inside the root and don't need these abs_srcdir kind of things at all.